repo_commit.go

```
@@ -96,8 +96,8 @@ func (repo *Repository) getCommitIdOfPackedRef(refpath string) ([]byte, error) {
 	scan := bufio.NewScanner(f)
 
 	for scan.Scan() {
-		if strings.Contains(scan.Text(), refpath) {
-			return scan.Bytes(), nil
+		if refLine, err := parseRefLine(scan.Text()); err == nil && refLine.refpath == refpath {
```

Is it okay that the error gets lost?